### ISSUE TITLE
(Existential operator) ?? instead of ?

### ISSUE THREAD

**Issue by aseemk** [jashkenas/coffeescript#1288]
Hey there,

Very awesome work on CoffeeScript. I'm a big fan of JavaScript, but I admit I'm really enamored by the various features and fixes CoffeeScript gives to the semantics of JS.

An example of a feature I love is the [existential operator](http://jashkenas.github.com/coffee-script/#operators) and its variants.

```
if foo? then alert 'foo has been set' else alert 'foo has not been set'
```

`?` immediately following a variable makes sense to me. However, this example did not:

```
footprints = yeti ? "bear"
```

Coming from JavaScript, since the `?` wasn't immediately following the variable (i.e. there's a space in between), it reads like the ternary operator; it read to me as "if yeti is set [i.e. not null or undefined], set footprints to 'bear', otherwise leave footprints as whatever it was [undefined by default]".

Since `?:` is used as a [ternary operator](http://en.wikipedia.org/wiki/Ternary_operator) in so many languages (JavaScript, Ruby, Perl, PHP, C/C++, Java, C#, ActionScript, ...), might I request changing the existential operator in this case to `??` instead? The only other languages that implement this operator (which they call the [null coalescing operator](http://en.wikipedia.org/wiki/Null_coalescing_operator)) use `??` as well: Perl and C#.

To recap, `foo?` would still be valid, I'm just requesting `foo ? bar` to instead be `foo ?? bar` to avoid confusion with the ternary operator from many other languages, improving readability. Thanks for your consideration, and great work again!

Cheers,
Aseem
